src/networking.c

```diff
@@ -2397,6 +2397,9 @@ int handleClientsWithPendingWrites(void) {
     listIter li;
     listNode *ln;
     listRewind(server.clients_pending_write, &li);
+    /* Avoid sending to I/O threads during out-of-memory conditions to free clients' COBs as quickly as possible */
+    int try_send_to_io_threads =
+        server.active_io_threads_num > 1 && (server.maxmemory == 0 || zmalloc_used_memory() <= server.maxmemory);
```

Comment:
I am not sure I like the fact that we keep our own logic here. there is a max memory logic encapsulated in getMaxmemoryState but I get it that it might be too expensive calling it here. maybe we should keep a maxmemory state indicator in the server which is updated in performEvictions or anytime we call getMaxmemoryState ?

Reply:
Thanks,  I've changed it to use `getMaxmemoryState`. The cost should be the same, except that `getMaxmemoryState `doesn't quickly return when `server.maxmemory` is 0. maybe we can change the `getMaxmemoryState` to do it.